src/Avalonia.FuncUI.Components/Context/Context.Hooks.fs

```diff
@@ -145,4 +145,15 @@ module IComponentContextExtensions =
         /// <param name="triggers">list of effect triggers</param>
         /// <param name="callerLineNumber">Usually provided by the compiler. Needs to be unique in the components context.</param>
         member this.useEffect (handler: unit -> unit, triggers: EffectTrigger list, [<CallerLineNumber>] ?callerLineNumber: int) =
-            this.useEffectHook (EffectHook.Create(HookIdentity.CallerCodeLocation callerLineNumber.Value, (fun _ -> handler(); null), triggers))
\ No newline at end of file
+            this.useEffectHook (EffectHook.Create(HookIdentity.CallerCodeLocation callerLineNumber.Value, (fun _ -> handler(); null), triggers))
+
+        /// Provides Elmish style updates for a writable model.
+        member this.useElmish<'Model, 'Msg> (writableModel: IWritable<'Model>, update) = 
+            let state = ElmishHook.ElmishState<'Model, 'Msg>(writableModel, update)
+            state.Model, state.Dispatch
+
+        /// Provides Elmish style updates for a model via the useState hook.
+        member this.useElmish<'Model, 'Msg> (init: 'Model, update) = 
+            let writableModel = this.useState (init, true)
+            let state = ElmishHook.ElmishState<'Model, 'Msg>(writableModel, update)
+            state.Model, state.Dispatch
```

Comment:
@JordanMarr it should be possible to move the Elmish related stuff to the FuncUI Elmish Project, right?

Reply:
I think it needs to live in `Avalonia.FuncUI.Components` because it depends on `useState` and `IWritable<'value>`.
It is designed to only work within component-based UIs.

Reply:
Adding a reference to the components project to the Elmish one should be fine for until we move components to the main FuncUI Project. 

This way all the Elmish stuff comes with the Elmish package. 

> It is designed to only work within component-based UIs.

Components will be there by default. Even when using Elmish for the larger part of the app having some state being local is super handy IMHO.